pyproject.toml

```diff
@@ -1,3 +1,94 @@
+[build-system]
+requires = ["flit_core >=3.2,<4"]
+build-backend = "flit_core.buildapi"
+
+[tool.flit.sdist]
+include = [
+    "AUTHORS.rst",
+    "CITATION.cff",
+    "CODE_OF_CONDUCT.md",
+    "CONTRIBUTING.rst",
+    "LICENSE",
+    "README.rst",
+    "VERSION",
+    "tox.ini",
+    "ci/",
+    "docs/",
+    "examples/",
+    "src/",
+    "tests/",
+    ".bumpversion.cfg",
+    ".coveragerc",
+    ".editorconfig",
+    ".flake8",
+    ".pep8speaks.yml",
+    ".readthedocs.yaml",
+    ".scrutinizer.yml",
+]
+exclude = ["docs/_build"]
+
+[project]
+name = "oemof.solph"
+dynamic = ["version"]
+description = "A model generator for energy system modelling and optimisation."
+readme = "README.rst"
+authors = [
+    {name = "oemof developer group", email = "contact@oemof.org"},
+]
+classifiers = [
+    "Development Status :: 5 - Production/Stable",
+    "Intended Audience :: Developers",
+    "Intended Audience :: Science/Research",
+    "License :: OSI Approved :: MIT License",
+    "Operating System :: Unix",
+    "Operating System :: POSIX",
+    "Operating System :: Microsoft :: Windows",
+    "Operating System :: MacOS",
+    "Operating System :: OS Independent",
+    "Programming Language :: Python",
+    "Programming Language :: Python :: 3",
+    "Programming Language :: Python :: 3.8",
+    "Programming Language :: Python :: 3.9",
+    "Programming Language :: Python :: 3.10",
```

Comment:
Tox tests Python 3.9, 3.10, and 3.11.

Reply:
The minimal supported Python version according to the specifications of the dependencies is 3.8 I just noticed. With the fix in the pandas H to h for the frequency (do not know how to search for merges on GH mobile app... It was the most recent PR merged to dev), that compatibility is actually broken via pandas. I am not sure, how the procedure is to change the minimum supported version of python.

Reply:
There is no real procedure. Also, technically we do not break compatibility by removing support for outdated versions of Python, we just don't provide new updates. (If Pandas 2.x requires Python 3.9+, we might include this change in #1051.)

Reply:
I did a bit of research:

- python 3.8 is supported until pandas 2.0.3 (we are at 2.2.2 currently)
- with python 3.8 and pandas 2.0.3 the function call of `pd.date_range()
  - **fails**: `pd.date_range("1/1/2024", periods=10, freq=f"0.5h")`
  - **succeeds**: `pd.date_range("1/1/2024", periods=10, freq=f"0.5h")`
- with python 3.9/3.10/3.11 and pandas 2.0.3 the function call of `pd.date_range()` also has the same behavior as the python 3.8 version
- only, with pandas>=2.2 the `0.5h` will start working (which is a pandas version not supported by python 3.8)

So what I actually did, when changing `"H"` to  `"h"` was under the hood require pandas >= 2.2, I think. How should we resolve this then? Requiring pandas>=2 I see no issue with in that context, we could even bring back python 3.8, but we would have to handle, how non-integer frequencies are dealt with by the `create_date_index` function.

Reply:
After a coffee, I suggest:

- we can leave python 3.8 support, but already prompt people that it is untested and will be removed with the next major release
- we come up with a fix for the `create_date_index` to work with pandas >= 2 but pandas < 2.2
- we require pandas >= 2

The pandas required stuff can be done in #1051. I will add the prompt for the depreciation of python3.8 in this pr.